shopfloor_single_product_transfer/tests/test_set_quantity.py

```diff
@@ -0,0 +1,442 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from .common import CommonCase
+
+
+class TestSetQuantity(CommonCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.location = cls.location_src
+        cls.product = cls.product_a
+        cls.packaging = cls.product_a_packaging
+        cls.packaging.qty = 5
+
+    @classmethod
+    def _setup_picking(cls, lot=None):
+        if lot:
+            cls._set_product_tracking_by_lot(cls.product)
+        cls._add_stock_to_product(cls.product, cls.location, 10, lot=lot)
+        return cls._create_picking(lines=[(cls.product, 10)])
+
+    @classmethod
+    def _setup_chained_picking(cls, picking):
+        next_moves = picking.move_lines.browse()
+        for move in picking.move_lines:
+            next_moves |= move.copy(
+                {
+                    "move_orig_ids": [(6, 0, move.ids)],
+                    "location_id": move.location_dest_id.id,
+                    "location_dest_id": cls.customer_location.id,
+                }
+            )
+        next_moves._assign_picking()
+        next_picking = next_moves.picking_id
+        next_picking.action_confirm()
+        next_picking.action_assign()
+        return next_picking
+
+    def test_set_quantity_barcode_not_found(self):
+        # First, select a picking
+        picking = self._setup_picking()
+        self.service.dispatch(
+            "scan_product",
+            params={"location_id": self.location.id, "barcode": self.product.barcode},
+        )
+        move_line = picking.move_line_ids
+        # Then try to scan an invalid barcode
+        response = self.service.dispatch(
+            "set_quantity", params={"selected_line_id": move_line.id, "barcode": "NOPE"}
+        )
+        expected_message = {"message_type": "error", "body": "Barcode not found"}
+        data = {
+            "move_line": self._data_for_move_line(move_line),
+            "asking_confirmation": False,
+        }
+        self.assert_response(
+            response, next_state="set_quantity", message=expected_message, data=data
+        )
+
+    def test_set_quantity_scan_product_prefill_qty_disabled(self):
```

Comment:
has the option is named starting with 'no'. The name of the test function should also to stay coherent
```suggestion
    def test_set_quantity_scan_product_no_prefill_qty_disabled(self):
```